app/src/main/java/by/itechart/android/ui/view/PyramidView.kt

```diff
@@ -0,0 +1,123 @@
+package by.itechart.android.ui.view
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.Path
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import android.widget.ImageView
+import by.itechart.android.R
+import by.itechart.android.ext.loadCircle
+import kotlinx.android.synthetic.main.view_pyramid.view.*
+import java.lang.IllegalArgumentException
+
+
+class PyramidView @JvmOverloads constructor(
+    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr) {
+
+    private lateinit var colors: IntArray
+
+    private val path = Path()
+    private val paint = Paint()
+
+    private val levelString: String
+
+    private var textSize: Float = 0f
+
+    private var blockHeight: Float = 0f
+    private var fillBlockHeight: Float = 0f
+    private var photoSize: Float = 0f
+
+    var levels: Int = 6
+        set(value) {
+            if (value < 0) throw IllegalArgumentException("Level count must be positive number")
+            field = value
+            invalidate()
+        }
+
+    var userLevel: Int = 3
+        set(value) {
+            if (value > levels) throw IllegalArgumentException("User level must be less or equal than level numbers")
+            if (value < 0) throw IllegalArgumentException("User level must be positive number")
+            field = value
+            pyramidImageView.updateVerticalPosition()
+        }
+
+    init {
+        setWillNotDraw(false)
+        inflate(context, R.layout.view_pyramid, this)
+        initAttributes(context, attrs)
+        levelString = context.resources.getString(R.string.community_level)
+        paint.textSize = textSize
+    }
+
+    override fun onSizeChanged(w: Int, h: Int, oldw: Int, oldh: Int) {
+        super.onSizeChanged(w, h, oldw, oldh)
+        blockHeight = height.toFloat() / levels
+        fillBlockHeight = blockHeight * 9f / 10
+        photoSize = fillBlockHeight * 4f / 5f
+    }
+
+    override fun onDraw(canvas: Canvas?) {
+        super.onDraw(canvas)
+        canvas?.let {
+            val pyramidCenterX = (width.toFloat() * 5f / 7f)
```

Comment:
remove ALL initializations from onDraw() method

onDraw(canvas: Canvas) {
    for (i in 0 until levels) { drawLevel() }
    drawAvatar()
}